dlib/dnn/utilities.h

```diff
@@ -301,6 +301,65 @@ namespace dlib
         return num_parameters;
     }
 
+// ----------------------------------------------------------------------------------------
+
+    namespace impl
+    {
+        class visitor_learning_rate_multiplier
+        {
+        public:
+            visitor_learning_rate_multiplier(double new_learning_rate_multiplier_) :
+                new_learning_rate_multiplier(new_learning_rate_multiplier_) { }
+
+            template <typename T>
+            void set_new_learning_rate_multiplier(T& l) const
+            {
+                set_learning_rate_multiplier(l, new_learning_rate_multiplier);
+                set_bias_learning_rate_multiplier(l, new_learning_rate_multiplier);
```

Comment:
The definition of the bias learning rate multiplier is this:
```
        double get_bias_learning_rate_multiplier(
        ) const;
        /*!
            ensures
                - returns a multiplier number.  The interpretation is that this object is
                  requesting that the learning rate used to optimize its bias parameters be
                  multiplied by get_learning_rate_multiplier()*get_bias_learning_rate_multiplier().
        !*/
```
So only call `set_learning_rate_multiplier`, otherwise the bias learning rate will not result in the desired value.

Reply:
Can I assume the same is true for the weight decay multiplier?

Reply:
Sorry I misunderstood you, fixed in new commit

Reply:
Yep, see the docs for that too https://github.com/davisking/dlib/blob/master/dlib/dnn/layers_abstract.h#L547.  It's the same deal.